backup.sh

```diff
@@ -1,145 +1,228 @@
+    exit 1
+  fi
+
+  if [ "${POSTGRES_PASSWORD}" = "**None**" -a "${POSTGRES_PASSWORD_FILE}" = "**None**" -a "${POSTGRES_PASSFILE_STORE}" = "**None**" ]; then
+    echo "You need to set the POSTGRES_PASSWORD or POSTGRES_PASSWORD_FILE or POSTGRES_PASSFILE_STORE environment variable or link to a container named POSTGRES."
+    exit 1
+  fi
+
+  #Process vars
+  if [ "${POSTGRES_DB_FILE}" = "**None**" ]; then
+    POSTGRES_DBS=$(echo "${POSTGRES_DB}" | tr , " ")
+  elif [ -r "${POSTGRES_DB_FILE}" ]; then
+    POSTGRES_DBS=$(cat "${POSTGRES_DB_FILE}")
+  else
+    echo "Missing POSTGRES_DB_FILE file."
+    exit 1
+  fi
+  if [ "${POSTGRES_USER_FILE}" = "**None**" ]; then
+    export PGUSER="${POSTGRES_USER}"
+  elif [ -r "${POSTGRES_USER_FILE}" ]; then
+    export PGUSER=$(cat "${POSTGRES_USER_FILE}")
+  else
+    echo "Missing POSTGRES_USER_FILE file."
+    exit 1
+  fi
+  if [ "${POSTGRES_PASSWORD_FILE}" = "**None**" -a "${POSTGRES_PASSFILE_STORE}" = "**None**" ]; then
+    export PGPASSWORD="${POSTGRES_PASSWORD}"
+  elif [ -r "${POSTGRES_PASSWORD_FILE}" ]; then
+    export PGPASSWORD=$(cat "${POSTGRES_PASSWORD_FILE}")
+  elif [ -r "${POSTGRES_PASSFILE_STORE}" ]; then
+    export PGPASSFILE="${POSTGRES_PASSFILE_STORE}"
+  else
+    echo "Missing POSTGRES_PASSWORD_FILE or POSTGRES_PASSFILE_STORE file."
+    exit 1
+  fi
+
+
+  # Pre-backup hook
+  if [ -d "${HOOKS_DIR}" ]; then
+    run-parts -a "pre-backup" --exit-on-error "${HOOKS_DIR}"
+  fi
+
+  #Initialize dirs
+  FREQUENCY=( last daily weekly monthly )
+
+  for f in ${FREQUENCY[@]}
+  do
+  
+    mkdir -p "${BACKUP_DIR}/${f}/"
+
+  done
+
+}
+
+#Create Backups
+create_backups () {
+
+  for DB in ${POSTGRES_DBS}
+  do
+
+    LAST_FILENAME="${DB}-`date +%Y%m%d-%H%M%S`${BACKUP_SUFFIX}"
+    FILE="${BACKUP_DIR}/last/${LAST_FILENAME}"
+
+    create_dump
+
+    echo "Point last backup file to this last backup..."
+    ln -svf "${LAST_FILENAME}" "${BACKUP_DIR}/last/${DB}-latest${BACKUP_SUFFIX}"
+
+    create_hardlinks "${FILE}" "daily"
+    create_hardlinks "${FILE}" "monthly"
+    create_hardlinks "${FILE}" "weekly"
+
+    echo "SQL backup created successfully"
+
+  done
+
+  # Post-backup hook
+  if [ -d "${HOOKS_DIR}" ]; then
+    run-parts -a "post-backup" --reverse --exit-on-error "${HOOKS_DIR}"
+  fi
+
+}
+
+#Create dump
+create_dump () {
+
   if [ "${POSTGRES_CLUSTER}" = "TRUE" ]; then
     echo "Creating cluster dump of ${DB} database from ${POSTGRES_HOST}..."
     pg_dumpall -l "${DB}" ${POSTGRES_EXTRA_OPTS} | gzip > "${FILE}"
   else
     echo "Creating dump of ${DB} database from ${POSTGRES_HOST}..."
     pg_dump -d "${DB}" -f "${FILE}" ${POSTGRES_EXTRA_OPTS}
   fi
+
+}
+
+create_hardlinks () {
+  
+  SRC=$1
+  INCREMENT=$2
+
+  if [ "${INCREMENT}" = "daily" ]
+  then
+
+    FILENAME="${DB}-`date +%Y%m%d`${BACKUP_SUFFIX}"
+
+  elif [ "${INCREMENT}" = "weekly" ]
+  then
+
+    FILENAME="${DB}-`date +%G%V`${BACKUP_SUFFIX}"
+
+  elif [ "${INCREMENT}" = "monthly"]
```

Comment:
Here there are a typo, and the CI is not failing.
Separate the setup function is making this kind of issues.

Reply:
There is an space missing, and it fails, look into the CI logs, you will see the script fails at this line.